internal/core/metadata/rest_deviceprofile.go

```diff
@@ -385,6 +385,12 @@ func restAddProfileByYaml(w http.ResponseWriter, r *http.Request) {
 		LoggingClient.Error(err.Error(), "")
 		return
 	}
+	if len(data) == 0 {
+		err := errors.New("YAML file is empty")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		LoggingClient.Error(err.Error(), "")
```

Comment:
I understand you're following along with the other statements in the file, but this is not how we want to do logging going forward.

The syntax for a logging statement is `(message string, args string...)` where message is a required message to the user and args is a collection of key/value pairs corresponding to some additional info you want to pass to the user. We're moving away from using `fmt.sprintf` and transitioning statements over to a string literal as the message with any variables as a value in a k/v.

I'd recommend updating this statement to
`LoggingClient.Error("failed to add profile by YAML", "error message", "err.Error()")`

If you're feeling saucy, it would be greatly appreciated if you did the same to more of the statements in the file. At the very least however we don't want to introduce more of the old style into the codebase.